runtime/obj.c

```diff
@@ -33,12 +33,15 @@
 
 static int obj_tag (value arg)
 {
+  header_t hd;
+
   if (Is_long (arg)){
     return 1000;   /* int_tag */
   }else if ((long) arg & (sizeof (value) - 1)){
     return 1002;   /* unaligned_tag */
   }else{
-    return Tag_val(arg);
+    hd = (header_t)atomic_load_acq(Hp_atomic_val(arg));
```

Comment:
```suggestion
    /* The acquire load ensures that reading the field of a Forward_tag
       block has appropriate synchronization. */
    hd = (header_t)atomic_load_acq(Hp_atomic_val(arg));
```

Reply:
Added comment in https://github.com/ocaml-multicore/ocaml-multicore/pull/760/commits/c45feb0653bf56adac0ccd5568b035b87b13d47b that also says where the field read occurs.